src/VisualStudio/TestUtilities/InProcess/InteractiveWindow_InProc.cs

```diff
@@ -0,0 +1,173 @@
+        /// </summary>
+        public string GetLastReplOutput()
+        {
+            // TODO: This may be flaky if the last submission contains ReplPromptText
+
+            var replText = GetReplTextWithoutPrompt();
+            int lastPromptIndex = replText.LastIndexOf(ReplPromptText);
+
+            replText = replText.Substring(lastPromptIndex, replText.Length - lastPromptIndex);
+            int lastSubmissionIndex = replText.LastIndexOf(ReplSubmissionText);
+
+            if (lastSubmissionIndex > 0)
+            {
+                replText = replText.Substring(lastSubmissionIndex, replText.Length - lastSubmissionIndex);
+            }
+            else if (!replText.StartsWith(ReplPromptText))
+            {
+                return replText;
+            }
+
+            int firstNewLineIndex = replText.IndexOf(Environment.NewLine);
+
+            if (firstNewLineIndex <= 0)
+            {
+                return replText;
+            }
+
+            firstNewLineIndex += Environment.NewLine.Length;
+            return replText.Substring(firstNewLineIndex, replText.Length - firstNewLineIndex);
+        }
+
+        public void Reset(bool waitForPrompt = true)
+        {
+            ExecuteCommand(ResetCommand);
+
+            if (waitForPrompt)
+            {
+                WaitForReplPrompt();
+            }
+        }
+
+        public void SubmitText(string text, bool waitForPrompt = true)
+        {
+            _interactiveWindow.SubmitAsync(new[] { text }).Wait();
+
+            if (waitForPrompt)
+            {
+                WaitForReplPrompt();
+            }
+        }
+
+        public void CloseWindow()
+        {
+            var dte = GetDTE();
+
+            foreach (EnvDTE.Window window in dte.Windows)
+            {
+                if (window.Caption == _windowTitle)
+                {
+                    window?.Close();
+                    break;
+                }
+            }
+        }
+
+        public void ShowWindow(bool waitForPrompt = true)
+        {
+            ExecuteCommand(_viewCommand);
+
+            if (waitForPrompt)
+            {
+                WaitForReplPrompt();
+            }
+        }
+
+        public void WaitForReplPrompt()
+        {
+            WaitForReplPromptAsync().ConfigureAwait(continueOnCapturedContext: false).GetAwaiter().GetResult();
+        }
+
+        private async Task WaitForReplPromptAsync()
+        {
+            while (!GetReplText().EndsWith(ReplPromptText))
+            {
+                await Task.Yield();
+            }
+        }
+
+        public void WaitForReplOutput(string outputText)
+        {
+            WaitForReplOutputAsync(outputText).ConfigureAwait(continueOnCapturedContext: false).GetAwaiter().GetResult();
+        }
+
+        private async Task WaitForReplOutputAsync(string outputText)
+        {
+            while (!GetReplText().EndsWith(outputText + Environment.NewLine + ReplPromptText))
```

Comment:
Also, can we not spin-wait in the first place.  Is there some reason we can't use the IAsyncListener approach to just 'await' when the window is available or not?